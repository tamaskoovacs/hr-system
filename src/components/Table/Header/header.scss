th {
  padding: 0.625em;
  text-align: center;

  font-size: 0.85em;
  letter-spacing: 0.1em;
  text-transform: uppercase;
}

@media screen and (max-width: 800px) {
  thead {
    border: none;
    clip: rect(0 0 0 0);
    height: 1px;
    margin: -1px;
    overflow: hidden;
    padding: 0;
    position: absolute;
    width: 1px;
  }
}
