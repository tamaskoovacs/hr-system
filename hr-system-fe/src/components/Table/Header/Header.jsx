import './header.scss'

const Header = ({ headers }) => {
  return (
    <thead>
      <tr>
        {headers?.map((header) => (<th key={header}>{header}</th>))}
      </tr>
    </thead>
  );
};

export default Header;
