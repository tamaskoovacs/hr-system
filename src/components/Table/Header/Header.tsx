import { HeaderProps } from 'interfaces/interfaces';
import './header.scss';

const Header: React.FC<HeaderProps> = ({ headers }) => {
  return (
    <thead>
      <tr>
        {headers?.map((header) => (
          <th key={header}>{header}</th>
        ))}
      </tr>
    </thead>
  );
};

export default Header;
