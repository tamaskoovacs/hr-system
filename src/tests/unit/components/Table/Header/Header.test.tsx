import React from 'react';
import { render, screen } from '@testing-library/react';
import Header from 'components/Table/Header/Header';

export const headers = [
  'Id',
  'First Name',
  'Last Name',
  'Email',
  'Date of birth',
  'Industry',
  'Salary',
  'Years of experience',
  'Actions'
];

describe('Header component', () => {
  test('It should matches snapshot', () => {
    const view = render(<Header headers={headers} />);
    expect(view.container).toMatchSnapshot();
  });
});
