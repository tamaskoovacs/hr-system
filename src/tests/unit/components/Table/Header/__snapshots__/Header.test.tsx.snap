// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component It should matches snapshot 1`] = `
<div>
  <thead>
    <tr>
      <th>
        Id
      </th>
      <th>
        First Name
      </th>
      <th>
        Last Name
      </th>
      <th>
        Email
      </th>
      <th>
        Date of birth
      </th>
      <th>
        Industry
      </th>
      <th>
        Salary
      </th>
      <th>
        Years of experience
      </th>
      <th>
        Actions
      </th>
    </tr>
  </thead>
</div>
`;
